site/src/pages/release-notes/1.27.0.mdx

```diff
@@ -0,0 +1,228 @@
+    .route()
+       .path("/api")
+       // Use apiServiceWorkerGroup for the API service.
+       .serviceWorkerGroup(apiServiceWorkerGroup)
+       .build(apiService)
+    .build();
+  ```
+- **Custom Exception Handling for gRPC clients**: You can customize how your gRPC clients respond to exceptions
+  by implementing <type://GrpcExceptionHandlerFunction>. #5347 #5351
+  ```java
+  GrpcClients
+    .builder("https://my-grpc-server.com") 
+    .exceptionHandler((ctx, cause, metadata) -> {
+      // Handle the exception.
+      return Status.INTERNAL.withDescription("Something went wrong");
+    })
+  ```
+- **Thrift Service Decorators**: You can now utilize <type://@Decorator> annotations to conveniently apply
+  custom logic to Thrift services. #4230 #5356
+  ```java
+  @Decorator(ClassDecorator.class)
+  class MyHelloService implements HelloService.AsyncIface {
+    @Decorator(MethodDecorator.class)
+    @Override
+    public void hello(String name, AsyncMethodCallback<String> handler) throws TException {
+      ...
+    }
+  }
+  ```
+- **Kubernetes Client Integration**: You can now use the popular [Fabric Kubernetes client](https://github.com/fabric8io/kubernetes-client)
+  seamlessly on top of Armeria when using the `armeria-kubernetes` module. #5167
+- **WebSocket Service Fallback**: Now, it is possible to set up a fallback service for 
+- <type://WebSocketService> to handle non-WebSocket requests seamlessly with
+  <type://WebSocketServiceBuilder#fallbackService(HttpService)> #5368
+  ```java
+  WebSocketService webSocketService = 
+    WebSocketService.builder()
+      // The fallback service is used when the request 
+      // is not a WebSocket request.
+      .fallbackService(httpService)
+      .build();
+  Server
+    .builder()
+    .service("/ws-or-http", webSocketService)
+    .build();
+  ```
+- **WebSocket Frame Aggregation**: You can now enable to aggregate WebSocket continuation frames for simplified 
+- message processing using <type://WebSocketServiceBuilder#aggregateContinuation(boolean)>. #5357
+  ```java
+  WebSocketService
+    .builder()
+    // Aggregate continuation frames into a single frame.
+    .aggregateContinuation(true)
+    .build(); 
+  ```
+- **Active Connection Monitoring**: You can track the number of active connections with 
+  `armeria.client.connections#value{state="active"}` for your clients when using 
+  <type://ConnectionPoolListener#metricCollecting(MeterRegistry)> #4277 #5288
+  ```java
+  ClientFactory clientFactory =
+    ClientFactory
+      .builder()
+      .connectionPoolListener(ConnectionPoolListener.metricCollecting(meterRegistry))
+      .build();
+  WebClient
+    .builder("https://my-backend.com")
+    .factory(clientFactory)
+    .build();
+  ```
+- **Graceful Executor Shutdown**: You can now gracefully terminate a custom `ExecutorService` in sync with the
+- <type://Server>'s shutdown procedure using <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)>
```

Comment:
```suggestion
  <type://Server>'s shutdown procedure using <type://ServerListenerBuilder#shutdownWhenStopped(ExecutorService,Duration)>
```

Reply:
Do you mean whenStopping?

Reply:
`shutdownWhenStopping` and `shutdownWhenStopped` is the actual API names.
Do you think we need to rename them before releasing 1.27.0?
https://github.com/line/armeria/blob/1862b7156c3606fcab7f2c09fa46f49588aae028/core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java#L327-L350

Reply:
What I meant was that you specified the `shutdownWhenStopped` twice in the original note. 😆 
The sentence is rewritten so it's okay now. 😉